source/kernel/src/services/sdmmc.rs

```diff
@@ -0,0 +1,467 @@
+//! SD/MMC Driver Service
+//!
+//! Driver for SD memory cards, SDIO cards and (e)MMC.
+//! This kernel driver will implement the actual protocol
+//! (which commands to send and how to interpret the response),
+//! while the platform driver will implement the device specific part
+//! (how to send and receive the data).
+#![warn(missing_docs)]
+use crate::{
+    comms::{
+        kchannel::{KChannel, KConsumer, KProducer},
+        oneshot::{self, Reusable},
+    },
+    mnemos_alloc::containers::FixedVec,
+    registry::{self, known_uuids, Envelope, KernelHandle, RegisteredDriver},
+    Kernel,
+};
+use core::{convert::Infallible, fmt, time::Duration};
```

Comment:
clippy is mad about a bunch of unused imports in this file:
```suggestion
use crate::{
    comms::oneshot::Reusable,
    mnemos_alloc::containers::FixedVec,
    registry::{self, known_uuids, Envelope, KernelHandle, RegisteredDriver},
    Kernel,
};
```